### ISSUE TITLE
[FR] Create LuaTexture with source

### ISSUE THREAD

**Issue by Krogoth100** [beyond-all-reason/spring#890]
Currently **gl.CreateTexture** accepts these parameters: (xsize, ysize [, zsize [, texProps]])

Proposition is to add this form: **gl.CreateTexture**(_source_ [,texProps])

_source_ parameter will be a string path of a texture file, to which the contents of a Lua texture _will be initialized to_.

Main purpose is to avoid micro-management of loading/copying named textures when you just need to load a texture from a file, but be able to operate on it as a Lua texture (via FBO and such).
Without this function, you have to use some redundant code, which also _slows game down_ (especially if we are talking work with big textures), because the texture is not directly initialized, as its best to do.

Once a texture is loaded via a simple one-liner, you will be able to work on it directly via a compute shader, and also _generate mipmaps using your own algorithm_ (via FBO with targetLod parameter). (See issue: https://github.com/beyond-all-reason/spring/issues/883)

The goal is to help make such Lua code as simple as possible.

**Comment by lhog** [comment]
I might miss something, but won't `gl.Texture("path_to_a_texture")` do what you want?

**Comment by Krogoth100** [comment]
> I might miss something, but won't `gl.Texture("path_to_a_texture")` do what you want?

Nope. Named textures are not allowed to render to (cannot be attachment for an FBO), nor they should be.
Need a way to initialize a **_Lua_** texture, that has many ways to operate on, with initial content from a file. _Directly_, without redundant binding said file texture and rendering to the Lua texture.
